src/pairs/mento-v2.ts

```diff
@@ -0,0 +1,255 @@
+    }
+    this.bucket0 = buckets.bucket0.decimalPlaces(0, 1);
+    this.bucket1 = buckets.bucket1.decimalPlaces(0, 1);
+  }
+
+  public swapExtraData(): string {
+    const broker = this.mento.getBroker();
+    return `${broker.address}${this.exchange.providerAddr.replace(
+      "0x",
+      ""
+    )}${this.exchange.id.replace("0x", "")}`;
+  }
+
+  public outputAmount(inputToken: Address, inputAmount: BigNumber) {
+    const spread = new BigNumber(this.poolExchange.config.spread.value._hex);
+    const getAmountOut = GET_AMOUNT_OUT[this.pricingModule];
+    const [tokenInBucketSize, tokenOutBucketSize] =
+      this.matchBuckets(inputToken);
+
+    const amountOut = getAmountOut(
+      tokenInBucketSize,
+      tokenOutBucketSize,
+      spread,
+      inputAmount
+    );
+    return amountOut;
+  }
+
+  public snapshot(): PairMentoV2Snapshot {
+    return {
+      swapFee: this.swapFee.toFixed(),
+      pricingModule: this.pricingModule,
+      bucket0: this.bucket0.toPrecision(),
+      bucket1: this.bucket1.toPrecision(),
+    };
+  }
+
+  public async restore(snapshot: PairMentoV2Snapshot): Promise<void> {
+    this.swapFee = new BigNumber(snapshot.swapFee);
+    this.pricingModule = snapshot.pricingModule;
+    this.bucket0 = new BigNumber(snapshot.bucket0);
+    this.bucket1 = new BigNumber(snapshot.bucket1);
+  }
+
+  private mentoBucketsAfterUpdate = async () => {
+    /*
+    Just like in mento.ts but use the PoolExchange in order to read:
+    lastBucketUpdate, updateFrequency and spread
+    also the bucket0 and bucket1.
+    Refreshed the PoolExchange structure first.
+    In the mentoBucketsAfter updated the referenced logic changes to:
+
+    function getUpdatedBuckets(PoolExchange memory exchange) internal view returns (uint256 bucket0, uint256 bucket1) {
+      bucket0 = exchange.config.stablePoolResetSize;
+      uint256 exchangeRateNumerator;
+      uint256 exchangeRateDenominator;
+      (exchangeRateNumerator, exchangeRateDenominator) = getOracleExchangeRate(exchange.config.referenceRateFeedID);
+  
+      bucket1 = exchangeRateDenominator.mul(bucket0).div(exchangeRateNumerator);
+    }
+    */
+    const bucket0 = new BigNumber(
+      this.poolExchange.config.stablePoolResetSize._hex
+    );
+    const [exchangeRateNumerator, exchangeRateDenominator] =
+      await this.getOracleExchangeRate(
+        this.poolExchange.config.referenceRateFeedID
+      );
+    const bucket1 = exchangeRateDenominator
+      .multipliedBy(new BigNumber(bucket0))
+      .dividedBy(exchangeRateNumerator);
+    return { bucket0, bucket1 };
+  };
+
+  protected getOracleExchangeRate = async (
+    rateFeedID: Address
+  ): Promise<[BigNumber, BigNumber]> => {
+    const sortedOracles = ISortedOracles__factory.connect(
+      this.sortedOraclesAddress,
+      this.provider
+    );
+    const [rateNumerator, rateDenominator] = await sortedOracles.medianRate(
+      rateFeedID
+    );
+    if (rateDenominator.lte(0))
```

Comment:
style nit: can you add brackets here. single line if statements with no brackets can be very confusing.